### PR TITLE
(launch) feat: implement update

### PR DESCRIPTION
**Pull request by mohandast52**
## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: apps/launch/components/MyStakingContracts/FieldConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    maxNumInactivityPeriods: {
+      rules: getGenericFieldRules(FieldConfig.maxNumInactivityPeriods.name),
+    },
+    livenessPeriod: {
+      rules: [
+        ...getGenericFieldRules(FieldConfig.livenessPeriod.name),
+        {
+          type: 'number',
+          min: 86400,
+          max: 86400 * 30,
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/FieldConfig.tsx:281]: I don't see any min/max rules for this field anywhere, could you please tell me where you get them from?

File: apps/launch/components/MyStakingContracts/FieldConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+          message: `Liveness period must be between 24 hours and 30 days`,
+        },
+      ],
+    },
+    timeForEmissions: {
+      rules: [
+        {
+          required: true,
+          message: `Please enter Time for emissions`,
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/FieldConfig.tsx:290]: can use getGenericFieldRules here?

File: apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+  const isLoading = useMemo(
+    () => isLivenessPeriodLoading || isMinimumStakingPeriodsLoading,
+    [isLivenessPeriodLoading, isMinimumStakingPeriodsLoading],
+  );
+  const data = useMemo(() => {
+    if (!livenessPeriod) return NA;
+    if (!minimumStakingPeriods) return NA;
+
+    const minStakingDuration = Number(minimumStakingPeriods) / Number(livenessPeriod);
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx:96]: I assume there's a confusion in naming here, should be `minimumStakingPeriods` = `minStakingDuration` / `livenessPeriod`

File: apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  const { data, isLoading } = useGetLivenessPeriod({ address });
+  return <ShowContent isLoading={isLoading} data={data ? `${data} seconds` : NA} />;
+};
+
+const TimeForEmissions: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useTimeForEmissions({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const AgentInstances: FC<{ address: Address }> = ({ address }) => {
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx:118]: can you please rename this to `NumAgentInstances` as well as the `useAgentInstances` hook? currently it sounds like it returns a list of instances

File: apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx (hunk 3 above)
Comment (Tanya-atatakai) [reply apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx:96]: yes, the name of the component is `MinimumStakingPeriods` but here we're calculating and returning minStakingDuration which is another variable

**Comment by kupermind** [comment]
@mohandast52 @Tanya-atatakai we are talking about time as hours, but setting the value in seconds. It didn't look nice when we tried to add (sec) to the Liveness period field, but something like this would be better. Otherwise we need to state that the period is at least 86400 seconds and at most 2... seconds.

**Review by kupermind — APPROVED** [review]

**Comment by oaksprout** [comment]
@mohandast52 looks good but see my suggestion above for the comment.

**Comment by Tanya-atatakai** [comment]
@mohandast52 I suggest we do the same for the `Time for emissions` field then, for consistency, because as Aleks mentioned it's also in seconds. So it'll be `Time for emissions (sec)`

**Review by Tanya-atatakai — APPROVED** [review]

**Review by oaksprout — APPROVED** [review]

**Pull request merged**